graphics/osl/Portfile

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           active_variants 1.1
+PortGroup           cmake 1.1
+
+github.setup        imageworks OpenShadingLanguage 1.10.9 Release-
+name                osl
+categories          graphics
+platforms           darwin
+license             BSD
+maintainers         @jasonliu-- openmaintainer
+homepage            http://opensource.imageworks.com
+description         advanced shading language for computer graphics \
+                    rendering
+long_description    Open Shading Language is a language developed by \
+                    Sony Pictures Imageworks, used for programming \
+                    shader effects in computer-generated imagery. It \
+                    can be used to define how surfaces or volumes \
+                    scatter light, and can be used for describing \
+                    materials, lights, displacement, and pattern \
+                    generation.
+
+checksums           rmd160  e03ba154633bdd2860e458e39818be2af00d1ee5 \
+                    sha256  6a62e89797c0dac382cb1816f87ae53deb332a4aedb89c956f05b36cd261a9a4 \
+                    size    13547442
+
+depends_build-append \
+                    port:flex \
+                    port:bison
+depends_lib         port:boost \
+                    port:ilmbase \
+                    port:pugixml \
+                    port:openimageio
+
+# OSL's tools require LLVM to be installed. It doesn't really care
+# which version, so we'll just go with the newest version of LLVM that
+# is already installed. If no LLVM is installed, we'll add the newest
+# version of LLVM as a dependency.
+set llvm_versions {9.0 8.0 7.0 6.0 5.0}
+set newest_llvm 9.0
+foreach ver ${llvm_versions} {
+    if {![catch {set result [active_variants llvm-$ver ""]}]} {
+        set llvm_version $ver
+        break
+    } else {
+        set llvm_version $newest_llvm
+    }
+}
+depends_lib-append  port:llvm-$llvm_version
```

Comment:
We generally disfavour this sort of opportunistic use-what-you-find behaviour, as it results in non deterministic builds, which can complicate debugging later on, for instance.

Instead please either just pick one version and use that, or if you think it useful offer choices in the form of variants, and make one of them, the most recent, the default.

Reply:
Okay, I've changed it so that the port simply sets a single version of LLVM. My every instinct as a programmer is screaming at me that hardcoding a value like this is bad programming style, but I guess many other portfiles also do this, so I'll just go with what others are doing.

Reply:
Normally i would agree with you, but portfiles are not really regular code but more prescriptions to build a specific package. Many things are hardcoded, names, repositories, versions, checksums etc. In that regard consistent repeatability is much more important.

If it really offends you, provide a choice via a set of variants, and then just make one the default.